# 📝 02 - Markdown and Formatting

Jupyter notebooks are not just for code — you can mix **Markdown** and code to create clear, professional-looking reports.

This notebook introduces:
- Writing Markdown cells (headings, lists, links)
- Adding inline **LaTeX math**
- Embedding images
- Why formatting matters in data science communication


## 1. Headings and Text Styles

### Example
# Heading 1
## Heading 2
### Heading 3

**Bold text**, *italic text*, `inline code`

✅ **Your Turn**: Create a cell with the heading `## My Favorite Data Tools` and list at least three tools in bold.

In [1]:
## My Favorite Data Tools

- **Python**
- **Jupyter Notebook**
- **GitHub**

SyntaxError: invalid syntax (3682648153.py, line 3)

## 2. Lists

### Example
- Item 1
- Item 2
  - Sub-item 2a
  - Sub-item 2b

1. First
2. Second
3. Third

✅ **Your Turn**: Make an ordered list of your top 3 programming languages.

In [ ]:
1. Python  
2. JavaScript  
3. C++

## 3. Links and Images

### Links
[OpenAI](https://www.openai.com)

### Images
![MSU Logo](https://upload.wikimedia.org/wikipedia/en/8/8b/Midwestern_State_University_seal.png)

✅ **Your Turn**: Insert a link to your favorite website and embed an image (can be a URL or local file).

In [4]:
![Visit YouTube](https://www.youtube.com)

![Sample Image](https://images.sidearmdev.com/convert?url=https%3a%2f%2fdxbhsrqyrr690.cloudfront.net%2fsidearm.nextgen.sites%2fmwsu.sidearmsports.com%2fimages%2f2025%2f9%2f12%2f25_9_6_FB_WCU_Captians__1_.jpg&type=webp)

/bin/bash: -c: line 1: syntax error near unexpected token `('
/bin/bash: -c: line 1: `[Visit YouTube](https://www.youtube.com)'
/bin/bash: -c: line 1: syntax error near unexpected token `('
/bin/bash: -c: line 1: `[Sample Image](https://images.sidearmdev.com/convert?url=https%3a%2f%2fdxbhsrqyrr690.cloudfront.net%2fsidearm.nextgen.sites%2fmwsu.sidearmsports.com%2fimages%2f2025%2f9%2f12%2f25_9_6_FB_WCU_Captians__1_.jpg&type=webp)'


## 4. LaTeX Math

You can render math using LaTeX syntax inside `$ ... $` for inline or `$$ ... $$` for block math.

Example:  
Inline: $E = mc^2$  
Block:
$$
\frac{\partial f}{\partial x} = 2x
$$

✅ **Your Turn**: Write the quadratic formula in LaTeX.

In [ ]:
$$
x = \frac{-b \pm \sqrt{b^2 - 4ac}}{2a}
$$

SyntaxError: invalid syntax (870890665.py, line 1)

## 5. Mixing Code and Markdown

In [3]:
# Example: create data in Python
numbers = [1, 2, 3, 4, 5]
squares = [n**2 for n in numbers]
squares

[1, 4, 9, 16, 25]

Now we explain results in Markdown:
- The list `numbers` contained 1–5.
- The list `squares` contains each number squared.
- Mixing narrative + results is why notebooks are so powerful.

✅ **Your Turn**: Create a list of 3 numbers, calculate their cubes in code, then explain in Markdown.

In [2]:
numbers = [2, 4, 6]
cubes = [n**3 for n in numbers]
cubes

[8, 64, 216]

The list `numbers` contains the values 2, 4, and 6.

The list `cubes` contains the cube of each number:
- 2³ = 8  
- 4³ = 64  
- 6³ = 216  

This demonstrates how we can compute values in code and explain them clearly using Markdown.

---
### Summary
- Use headings and formatting to organize.
- Lists and links make content readable.
- LaTeX lets you add professional math.
- Combine code + Markdown for clear data storytelling.
